web3/datastructures.py

```diff
@@ -21,30 +30,34 @@
 # https://docs.python.org/3/reference/datamodel.html#object.__hash__
 
 
-class ReadableAttributeDict(Mapping):
+pretty = TypeVar("pretty", bound=Any)
+
+
+class ReadableAttributeDict(Mapping[Any, Any]):
```

Comment:
I think the more correct way to do this would be:

```python
TKey = TypeVar("TKey", bound=Hashable)
TValue = TypeVar("TValue")

class ReadableAttributeDict(Mapping[TKey, TValue]):
    ...
```

This does require that you specify key/value types in most places that you use this but I think that's a good thing.